src/flp/szk.rs

```diff
@@ -276,24 +240,81 @@ impl<F: FieldElement, const SEED_SIZE: usize> Encode for SzkVerifier<F, SEED_SIZ
 }
 
 impl<F: FieldElement + Decode, const SEED_SIZE: usize> ParameterizedDecode<(bool, usize)>
-    for SzkVerifier<F, SEED_SIZE>
+    for SzkQueryShare<F, SEED_SIZE>
 {
     fn decode_with_param(
         (requires_joint_rand, verifier_len): &(bool, usize),
         bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        Ok(SzkQueryShare {
+            joint_rand_part_opt: (*requires_joint_rand)
+                .then(|| Seed::<SEED_SIZE>::decode(bytes))
+                .transpose()?,
+            flp_verifier: decode_fieldvec(*verifier_len, bytes)?,
+        })
+    }
+}
+
+/// Szk query state.
+///
+/// The state that needs to be stored by an Szk verifier between query() and decide().
+pub type SzkQueryState<const SEED_SIZE: usize> = Option<Seed<SEED_SIZE>>;
+
+/// Verifier type for the SZK proof.
+pub type SzkVerifier<F> = Vec<F>;
+
+impl<F: FieldElement> Encode for SzkVerifier<F> {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        encode_fieldvec(self, bytes)?;
+        Ok(())
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        Some(self.len() * F::ENCODED_SIZE)
+    }
+}
+
+impl<F: FieldElement + Decode> ParameterizedDecode<usize> for SzkVerifier<F> {
+    fn decode_with_param(
+        verifier_len: &usize,
+        bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        decode_fieldvec(*verifier_len, bytes)
+    }
+}
+
+/// Joint share type for the SZK proof.
```

Comment:
I think this needs some further explanation, since this is a novel term.
```suggestion
/// Joint share type for the SZK proof.
///
/// This is produced by [`Szk::merge_verifiers`] as the result of combining two query shares.
/// It contains the joint randomness parts from each aggregator, if applicable. It is consumed by [`Szk::decide`].
```

Or, taking https://github.com/divviup/libprio-rs/pull/1107/files#r1823527498 into account:
```suggestion
/// Joint share type for the SZK proof.
///
/// This is produced by [`Szk::merge_verifiers`] as the result of combining two query shares.
/// It contains the re-computed joint randomness seed, if applicable. It is consumed by [`Szk::decide`].
```